packages/tupaia-web-server/src/routes/MapOverlaysRoute.ts

```diff
@@ -61,46 +60,10 @@ export class MapOverlaysRoute extends Route<MapOverlaysRequest> {
       };
     }
 
-    // Map overlay groups can be nested so we need to keep
-    // searching until we find the root groups
-    let mapOverlayRelations = await ctx.services.central.fetchResources(
-      'mapOverlayGroupRelations',
-      {
-        filter: {
-          child_type: 'mapOverlay',
-          child_id: mapOverlays.map((overlay: MapOverlay) => overlay.id),
-        },
-        pageSize: DEFAULT_PAGE_SIZE,
-      },
-    );
-    let parentMapOverlayRelations = await ctx.services.central.fetchResources(
-      'mapOverlayGroupRelations',
-      {
-        filter: {
-          child_type: 'mapOverlayGroup',
-          child_id: mapOverlayRelations.map(
-            (relation: MapOverlayGroupRelation) => relation.map_overlay_group_id,
-          ),
-        },
-        pageSize: DEFAULT_PAGE_SIZE,
-      },
+    // Breaking orchestration server convention and accessing the db directly
+    const mapOverlayRelations = await this.req.models.mapOverlayGroupRelation.findParentRelationTree(
```

Comment:
I've left the final fetch for the `mapOverlayGroups` themselves as a request to `central-server`. I'd like to get this tested with this first section in place before deciding whether to replace that request with a db call too. Trying to avoid _too_ much breaking of orchestration server practices (at least without explicit reason)

Reply:
Good call. Arguably if we were to make a rule about this we might want to say that it's the central-server that enforces permissions, so we can't fetch resources themselves directly from the db and return them to the user in an orchestration server.